ui/app/AppLayouts/Wallet/views/AssetsDetailView.qml

```diff
@@ -241,17 +227,33 @@ Item {
                 active: RootStore.marketHistoryIsLoading
             }
 
+            function updateBalanceStore() {
+                let selectedTimeRangeEnum = balanceStore.timeRangeStrToEnum(graphDetail.selectedTimeRange)
+
+                let currencySymbol = RootStore.currencyStore.currentCurrency
+                if(!balanceStore.hasData(root.address, token.symbol, currencySymbol, selectedTimeRangeEnum)) {
+                    RootStore.fetchHistoricalBalanceForTokenAsJson(root.address, token.symbol, currencySymbol, selectedTimeRangeEnum)
+                }
+            }
+
             TokenBalanceHistoryStore {
                 id: balanceStore
 
-                address: root.address
-
-                onNewDataReady: (timeRange) => {
-                    let selectedTimeRange = timeRangeStrToEnum(graphDetail.selectedTimeRange)
-                    if (timeRange === selectedTimeRange && address === root.address) {
+                onNewDataReady: (address, tokenSymbol, currencySymbol, timeRange) => {
+                    if (timeRange === timeRangeStrToEnum(graphDetail.selectedTimeRange)) {
                         chart.updateToNewData()
                     }
                 }
+
+                Connections {
+                    target: root
+                    onAddressChanged: graphDetail.updateBalanceStore()
```

Comment:
On Qt5.15 this will raise a warning that `onFoo` handlers are deprecated. Need to use `function onFoo` instead.
```suggestion
                    function onAddressChanged() { graphDetail.updateBalanceStore() }
```